src/core/qgstiledownloadmanager.h

```diff
@@ -53,11 +53,18 @@ class CORE_EXPORT QgsTileDownloadManagerReply : public QObject
     bool hasFinished() const { return mHasFinished; }
     //! Returns binary data returned in the reply (only valid when already finished)
     QByteArray data() const { return mData; }
+    //! Returns the reply URL
+    QUrl url() const { return mUrl; }
+    //! Returns the attribute associated with the \a code
+    QVariant attribute( QNetworkRequest::Attribute code );
+    //! Returns the value of the known header \a header.
+    QVariant header( QNetworkRequest::KnownHeaders header );
+    //! Returns a list of raw header pairs
+    const QList<QNetworkReply::RawHeaderPair> rawHeaderPairs() const { return mRawHeaderPairs; }
```

Comment:
Just curious - were these changes necessary for something, or have you added them just in case? They do not seem to be used anywhere in the client code...

Reply:
Yeah, QgsWmsTiledImageDownloadHandler::tileReplyFinished() makes use of all these.

Reply:
Ahh right, good point, it just didn't show up in the diff and I got confused :-)